api/users/serializers.py

```diff
@@ -117,6 +123,25 @@ def get_absolute_url(self, obj):
             'version': self.context['request'].parser_context['kwargs']['version']
         })
 
+    def get_node_count(self, obj):
+        return default_node_list_queryset(model_cls=Node).filter(contributor__user__id=obj.id).count()
+
+    def get_quickfiles_count(self, obj):
+        return QuickFilesNode.objects.get(contributor__user__id=obj.id).files.filter(type='osf.osfstoragefile').count()
+
+    def get_registration_count(self, obj):
+        return default_node_list_queryset(model_cls=Registration).filter(contributor__user__id=obj.id).count()
```

Comment:
Same thing here, logged in user isn't accounted for in this query, only target user.  You are looking to see registrations where the target user is a contributor (which is correct), but not whether the logged in user can see those registrations.